src/azure-cli/azure/cli/command_modules/botservice/custom.py

```diff
@@ -74,15 +75,27 @@ def __prepare_configuration_file(cmd, resource_group_name, kudu_client, folder_p
             f.write(json.dumps(existing))
 
 
-def create(cmd, client, resource_group_name, resource_name, kind, msa_app_id, password, language=None,  # pylint: disable=too-many-locals, too-many-statements
+def create(cmd, client, resource_group_name, resource_name, kind, msa_app_id, password=None, language=None,  # pylint: disable=too-many-locals, too-many-statements, inconsistent-return-statements
            description=None, display_name=None, endpoint=None, tags=None, location='Central US',
-           sku_name='F0', version='v4', deploy_echo=None):
+           sku_name='F0', deploy_echo=None):
     # Kind parameter validation
     kind = kind.lower()
     registration_kind = 'registration'
     bot_kind = 'bot'
     webapp_kind = 'webapp'
 
+    # Mapping: registration is deprecated, we now use 'bot' kind for registration bots
```

Comment:
When do you deprecate registration bots? Can we show up warning message here to inform users such change?

Reply:
We haven't deprecated the "Bot Channels Registration" Azure Resource, the `kind` was changed over a year ago, and to preserve the current flow for users we just mapped `'registration'` to the correct value `'bot'`. 

This new line isn't new code, it's just restructured around the new name availability check. ([link to original location](https://github.com/Azure/azure-cli/pull/11318/files#diff-29e4a6bde35880c6917ba0382e947063L108-L111)).

I think `'registration'` is a clearer indicator of what the user is signing up for in regards to the Azure Bot Service, as opposed to `'bot'`, which is redundant and not clear.

If it is required, I can change the enum values to `'bot'` and `'webapp'`, in the [params](https://github.com/Azure/azure-cli/pull/11318/files#diff-9f85bf3488eb6b9381fd84553e8ef64aR58) and start the deprecation process for the `'registration'` value.

```py
c.argument('kind', options_list=['--kind', '-k'], arg_type=get_enum_type(['registration', 'webapp']), help='The kind of the bot.')
```